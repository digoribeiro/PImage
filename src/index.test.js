import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import EnzymeAdapter from 'enzyme-adapter-react-16';

import PImage from '.';

Enzyme.configure({ adapter: new EnzymeAdapter() });

describe('PImage', () => {
    const srcs = [
        'http://lorempixel.com/100/100/animals/8',
        'http://lorempixel.com/200/200/animals/8',
        'http://lorempixel.com/800/800/animals/8',
        'http://lorempixel.com/1200/1200/animals/8',
        'http://lorempixel.com/1600/1600/animals/8',
        'http://lorempixel.com/1800/1800/animals/8',
    ];

    it('should render with a single src', () => {
        const wrapper = shallow(<PImage srcs={[srcs[5]]}/>);
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('should render with multiple srcs', () => {
        const wrapper = shallow(<PImage srcs={srcs}/>);
        expect(wrapper.html()).toMatchSnapshot();
    });
});
