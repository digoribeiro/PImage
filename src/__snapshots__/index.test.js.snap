// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PImage should render with a single src 1`] = `"<img style=\\"width:100%;heigth:100%\\" src=\\"http://lorempixel.com/1800/1800/animals/8\\"/>"`;

exports[`PImage should render with multiple srcs 1`] = `"<img style=\\"width:100%;heigth:100%\\" src=\\"http://lorempixel.com/100/100/animals/8\\"/>"`;
